src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -2257,110 +2261,7 @@ namespace SurfaceGeometry {
             }
         }
 
-        //**********************************************************************************
-        //   Set up Zone Surface Pointers
```

Comment:
The methodology here for setting `*SurfaceFirst/Last` for zone and spaces was flawed, especially if a space had no surfaces. This has been redone and move to a new function `setSurfaceFirstLast`.